kubernetes-gcp-go/main.go

```diff
@@ -0,0 +1,152 @@
+		if err != nil {
+			gcpRegion = "us-central1"
+		}
+		// Get some additional configuration values or use defaults
+		cfg := config.New(ctx, "")
+		nodePoolCount, err := cfg.TryInt("nodePoolCount")
+		if err != nil {
+			nodePoolCount = 1
+		}
+
+		// Create a new network
+		gkeNetwork, err := compute.NewNetwork(ctx, "gke-network", &compute.NetworkArgs{
+			AutoCreateSubnetworks: pulumi.Bool(false),
+			Description:           pulumi.String("A virtual network for your GKE cluster(s)"),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create a subnet in the network
+		gkeSubnet, err := compute.NewSubnetwork(ctx, "gke-subnet", &compute.SubnetworkArgs{
+			IpCidrRange:           pulumi.String("10.128.0.0/12"),
+			Network:               gkeNetwork.ID(),
+			PrivateIpGoogleAccess: pulumi.Bool(true),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create a new GKE cluster
+		gkeCluster, err := container.NewCluster(ctx, "gke-cluster", &container.ClusterArgs{
+			AddonsConfig: &container.ClusterAddonsConfigArgs{
+				DnsCacheConfig: &container.ClusterAddonsConfigDnsCacheConfigArgs{
+					Enabled: pulumi.Bool(true),
+				},
+			},
+			BinaryAuthorization: &container.ClusterBinaryAuthorizationArgs{
+				EvaluationMode: pulumi.String("PROJECT_SINGLETON_POLICY_ENFORCE"),
+			},
+			DatapathProvider: pulumi.String("ADVANCED_DATAPATH"),
+			Description:      pulumi.String("A GKE cluster"),
+			InitialNodeCount: pulumi.Int(1),
+			IpAllocationPolicy: &container.ClusterIpAllocationPolicyArgs{
+				ClusterIpv4CidrBlock:  pulumi.String("/14"),
+				ServicesIpv4CidrBlock: pulumi.String("/20"),
+			},
+			Location: pulumi.String(gcpRegion),
+			MasterAuthorizedNetworksConfig: &container.ClusterMasterAuthorizedNetworksConfigArgs{
+				CidrBlocks: container.ClusterMasterAuthorizedNetworksConfigCidrBlockArray{
+					&container.ClusterMasterAuthorizedNetworksConfigCidrBlockArgs{
+						CidrBlock:   pulumi.String("0.0.0.0/0"),
+						DisplayName: pulumi.String("All networks"),
+					},
+				},
+			},
+			Network:        gkeNetwork.Name,
+			NetworkingMode: pulumi.String("VPC_NATIVE"),
+			PrivateClusterConfig: &container.ClusterPrivateClusterConfigArgs{
+				EnablePrivateNodes:    pulumi.Bool(true),
+				EnablePrivateEndpoint: pulumi.Bool(false),
+				MasterIpv4CidrBlock:   pulumi.String("10.100.0.0/28"),
+			},
+			RemoveDefaultNodePool: pulumi.Bool(true),
+			ReleaseChannel: &container.ClusterReleaseChannelArgs{
+				Channel: pulumi.String("STABLE"),
+			},
+			Subnetwork: gkeSubnet.Name,
+			WorkloadIdentityConfig: &container.ClusterWorkloadIdentityConfigArgs{
+				WorkloadPool: pulumi.String(fmt.Sprintf("%v.svc.id.goog", gcpProject)),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create a GCP Service Account for the node pool
+		gkeNodepoolSa, err := serviceaccount.NewAccount(ctx, "gke-nodepool-sa", &serviceaccount.AccountArgs{
+			AccountId:   pulumi.String("nodepool-1-sa"),
+			DisplayName: pulumi.String("Nodepool 1 Service Account"),
```

Comment:
If we do choose to change the way we name the account ID, we might want to make this something more generic (or just leave it out, unless that makes things too weird in the GCP console).

Reply:
In 440fb15, I added the uniquely-named cluster name to the name of the node pool SA, which guarantees unique-ness.